core/dbt/plugins/manager.py

```diff
@@ -63,6 +65,17 @@ def get_manifest_artifacts(self, manifest: Manifest) -> PluginArtifacts:
         raise NotImplementedError(f"get_manifest_artifacts hook not implemented for {self.name}")
 
 
+@functools.cache
```

Comment:
@peterallenwebb Per [docs](https://docs.python.org/3/library/functools.html#functools.cache), sounds like `@functools.cache` might just be 3.9+ syntactic sugar for `@functools.lru_cache(maxsize=None)`, which is available on earlier versions?